modules/imap/handler_modules.php

```diff
@@ -252,18 +252,37 @@ class Hm_Handler_imap_process_move extends Hm_Handler_Module {
     public function process() {
         list($success, $form) = $this->process_form(array('imap_move_to', 'imap_move_page', 'imap_move_action', 'imap_move_ids'));
         if ($success) {
+            $screen = false;
+            $parts = explode("_", $this->request->get['list_path']);
+            $imap_server_id = $parts[1] ?? '';
+            $cache = Hm_IMAP_List::get_cache($this->cache, $imap_server_id);
+            $imap = Hm_IMAP_List::connect($imap_server_id, $cache);
+            if ($form['imap_move_action'] == "screen_mail") {
+                $form['imap_move_action'] = "move";
+                $screen = true;
+                $screen_folder = 'Screen emails';
+                if (!count($imap->get_mailbox_status($screen_folder))) {
+                    $imap->create_mailbox($screen_folder);
+                }
+                $form['imap_move_to'] = $parts[0] ."_". $parts[1] ."_".bin2hex($screen_folder);
+            }
+
             list($msg_ids, $dest_path, $same_server_ids, $other_server_ids) = process_move_to_arguments($form);
             $moved = array();
             if (count($same_server_ids) > 0) {
-                $moved = array_merge($moved, imap_move_same_server($same_server_ids, $form['imap_move_action'], $this->cache, $dest_path));
+                $moved = array_merge($moved, imap_move_same_server($same_server_ids, $form['imap_move_action'], $this->cache, $dest_path, $screen));
             }
             if (count($other_server_ids) > 0) {
                 $moved = array_merge($moved, imap_move_different_server($other_server_ids, $form['imap_move_action'], $dest_path, $this->cache));
 
             }
             if (count($moved) > 0 && count($moved) == count($msg_ids)) {
                 if ($form['imap_move_action'] == 'move') {
-                    Hm_Msgs::add('Messages moved');
+                    if ($screen) {
+                        Hm_Msgs::add('Email.s moved to Sreen email folder');
```

Comment:
Please correct the message - dot seems wrong and also the folder is called "Screen emails" - probably worth making the name configurable by user?